### PR TITLE
[Experiment] replace usage of `dangerouslySetInnerHTML` in `octicons-react`

### PR DESCRIPTION
**Pull request by joshblack**
## Background

https://github.com/primer/octicons/issues/796

I was looking into `dangerouslySetInnerHTML` usage in Primer and wanted to try an experiment with `@primer/octicons-react` that replaced the usage of `dangerouslySetInnerHTML` with JSX elements. This is being opened as a draft to gauge if the approach aligns with any existing efforts in this space 👀 

## What's changed

At a high-level, this PR changes a component from being defined as:

```jsx
function AccessibilityIcon(props) {
  var svgDataByHeight = { "16": { "width": 16, "path": "<path fill-rule=\"evenodd\" d=\"M9.923 5.302a3 3 0 10-3.847 0A2.713 2.713 0 005.9 5.5H2A.75.75 0 002 7h3.3l-.578 5.163-.362 2.997a.75.75 0 101.49.18L6.132 13h3.736l.282 2.34a.75.75 0 101.49-.18l-.362-2.997L10.7 7H14a.75.75 0 000-1.5h-3.899a2.697 2.697 0 00-.178-.198zM9.5 3a1.5 1.5 0 11-3 0 1.5 1.5 0 013 0zm-.3 4.073l.495 4.427h-3.39l.496-4.427a1.207 1.207 0 012.398 0z\"></path>" } };
  return React.createElement('svg', getSvgProps(_extends({}, props, { svgDataByHeight: svgDataByHeight })));
}

AccessibilityIcon.defaultProps = {
  className: 'octicon octicon-accessibility',
  size: 16,
  verticalAlign: 'text-bottom'
};
```

To the following:

```jsx
const AccessibilityIcon = createIconComponent("AccessibilityIcon", "octicon octicon-accessibility", {
  "16": {
    "width": 16,
    "path": /*#__PURE__*/React.createElement("path", {
      fillRule: "evenodd",
      d: "M9.923 5.302a3 3 0 10-3.847 0A2.713 2.713 0 005.9 5.5H2A.75.75 0 002 7h3.3l-.578 5.163-.362 2.997a.75.75 0 101.49.18L6.132 13h3.736l.282 2.34a.75.75 0 101.49-.18l-.362-2.997L10.7 7H14a.75.75 0 000-1.5h-3.899a2.697 2.697 0 00-.178-.198zM9.5 3a1.5 1.5 0 11-3 0 1.5 1.5 0 013 0zm-.3 4.073l.495 4.427h-3.39l.496-4.427a1.207 1.207 0 012.398 0z"
    })
  }
});
```

Under the hood, `createIconComponent` returns the `AccessibilityIcon` component. This function brings over the existing `getSvgProps` behavior and bakes it into the returned component.

[`createIconComponent`](https://github.com/primer/octicons/pull/837/files#diff-5e0add2e1cec1673d52c08a8090fae3f9e1adca177dcfe0a5d5c292795185bf7R48)

**How it works**

The default `lib/build/data.json` file now includes an AST of the SVG asset. The icon builder in `lib/octicons_react` converts this svg AST to JS which is included in the `svgData` argument of `createIconComponent`.

`createIconComponent` creates the component used by React and inlines the existing logic from `getSvgData` into the component.

The notable change at this point is that the `path` value is no longer a `string` but is instead a `JSXElement` (or `JSXFragment`) which can be set as the `children` value of the `<svg>` element.

## Testing & Reviewing

I updated the Next.js project to work with the package format so it should allow a preview of the new icon components!

## Feedback

I would love to hear what others think about this approach! I didn't see any other existing efforts when looking around but if there any let me know and I can close this PR 👀

### CODE REVIEW

File: lib/octicons_react/script/build.js [hunk 1]
```diff
@@ -16,17 +18,31 @@ function pascalCase(str) {
...
-
-${name}.defaultProps = {
-  className: 'octicon octicon-${key}',
-  size: 16,
-  verticalAlign: 'text-bottom'
-}
-`
+    const svgData = t.objectExpression(
+      Object.entries(octicon.heights).map(([height, icon]) => {
+        return t.objectProperty(
```
Comment (colebemis) [inline lib/octicons_react/script/build.js:23]: Love the you replaced the string literals with an AST 😍

File: lib/octicons_react/src/get-svg-props.js [hunk 2]
```diff
@@ -42,3 +44,45 @@ function closestNaturalHeight(naturalHeights, height) {
     .map(naturalHeight => parseInt(naturalHeight, 10))
     .reduce((acc, naturalHeight) => (naturalHeight <= height ? naturalHeight : acc), naturalHeights[0])
 }
+
+export function createIconComponent(name, defaultClassName, svgDataByHeight) {
```
Comment (colebemis) [inline lib/octicons_react/src/get-svg-props.js:48]: Can we remove the `getSvgProps` function?

**Review by colebemis — APPROVED** [review]
Love this approach. Nice work @joshblack!

**Comment by joshblack** [comment]
Splitting this up into: https://github.com/primer/octicons/pull/838 which should carry with it some of the dep updates plus tests to make sure no regressions happen 

Will then follow-up with a `createIconComponent` PR afterwards and run it against the tests setup in #838

**Comment by joshblack** [comment]
Closing in favor of: https://github.com/primer/octicons/pull/842

**Pull request closed without merging**